extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/KafkaStreamsTopologyManager.java

```diff
@@ -105,9 +104,9 @@ private static Properties getStreamsProperties(Properties properties, String boo
         streamsProperties.put(StreamsConfig.APPLICATION_ID_CONFIG, runtimeConfig.applicationId);
 
         // app id
-        if (runtimeConfig.applicationServer.isPresent()) {
-            streamsProperties.put(StreamsConfig.APPLICATION_SERVER_CONFIG, runtimeConfig.applicationServer.get());
-        }
+        streamsProperties.put(StreamsConfig.APPLICATION_SERVER_CONFIG,
+                runtimeConfig.applicationServer
+                        .orElseGet(() -> String.format("%s:%s", getQualifiedHostName(), getPort(runtimeConfig))));
```

Comment:
Minor gripe: we try to avoid lambdas in runtime code. It consumes memory and they are usually not that needed.

Reply:
Changed it. Thanks!